File: TOMIMSV4/Entity/Transacciones/Picking/Trans_Picking_Det/clsBeTrans_picking_det.vb
Public Class clsBeTrans_picking_det
    Implements ICloneable
    Implements IDisposable

    Public Property IdPickingDet() As Integer = 0
    Public Property IdPickingEnc() As Integer = 0
    Public Property IdPedidoDet() As Integer = 0
    Public Property IdOperadorBodega() As Integer = 0
    Public Property Cantidad() As Double = 0.0
    Public Property Cliente_dias() As Integer = 0
    Public Property Cantidad_recibida() As Double = 0.0
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False

    Sub New()
    End Sub

    Sub New(ByRef IdPickingDet As Integer, ByVal IdPickingEnc As Integer, ByVal IdPedidoDet As Integer, ByVal IdOperadorBodega As Integer, ByVal cantidad As Double, ByVal cliente_dias As Integer, ByVal cantidad_recibida As Double, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdPickingDet = IdPickingDet
        Me.IdPickingEnc = IdPickingEnc
        Me.IdPedidoDet = IdPedidoDet
        Me.IdOperadorBodega = IdOperadorBodega
        Me.Cantidad = Cantidad
        Me.Cliente_dias = Cliente_dias
        Me.Cantidad_recibida = Cantidad_recibida
        Me.User_agr = User_agr
        Me.Fec_agr = Fec_agr
        Me.User_mod = User_mod
        Me.Fec_mod = Fec_mod
        Me.Activo = Activo
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMSWebAPI/Entity/Picking/clsBeTrans_picking_det.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeTrans_picking_det : ICloneable
    {
        [Column("IdPickingDet")]
        [DisplayName("IdPickingDet")]
        public int IdPickingDet { get; set; } = 0;

        [Column("IdPickingEnc")]
        [DisplayName("IdPickingEnc")]
        public int IdPickingEnc { get; set; } = 0;

        [Column("IdPedidoEnc")]
        [DisplayName("IdPedidoEnc")]
        public int IdPedidoEnc { get; set; } = 0;

        [Column("IdPedidoDet")]
        [DisplayName("IdPedidoDet")]
        public int IdPedidoDet { get; set; } = 0;

        [Column("IdOperadorBodega")]
        [DisplayName("IdOperadorBodega")]
        public int IdOperadorBodega { get; set; } = 0;

        [Column("cantidad")]
        [DisplayName("cantidad")]
        public double Cantidad { get; set; } = 0;

        [Column("cliente_dias")]
        [DisplayName("cliente_dias")]
        public int Cliente_dias { get; set; } = 0;

        [Column("cantidad_recibida")]
        [DisplayName("cantidad_recibida")]
        public double Cantidad_recibida { get; set; } = 0;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("codigo")]
        [DisplayName("codigo")]
        public string Codigo { get; set; } = "";

        [Column("nombre")]
        [DisplayName("nombre")]
        public string Nombre { get; set; } = "";

        public clsBeTrans_picking_det() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
